flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/strategy/AdaptiveBroadcastJoinOptimizationStrategy.java

```diff
@@ -0,0 +1,213 @@
+import org.apache.flink.runtime.scheduler.adaptivebatch.BlockingResultInfo;
+import org.apache.flink.runtime.scheduler.adaptivebatch.OperatorsFinished;
+import org.apache.flink.streaming.api.graph.StreamGraphContext;
+import org.apache.flink.streaming.api.graph.util.ImmutableStreamEdge;
+import org.apache.flink.streaming.api.graph.util.ImmutableStreamNode;
+import org.apache.flink.streaming.api.graph.util.StreamEdgeUpdateRequestInfo;
+import org.apache.flink.streaming.runtime.partitioner.BroadcastPartitioner;
+import org.apache.flink.streaming.runtime.partitioner.ForwardPartitioner;
+import org.apache.flink.streaming.runtime.partitioner.StreamPartitioner;
+import org.apache.flink.table.api.config.OptimizerConfigOptions;
+import org.apache.flink.table.runtime.operators.join.FlinkJoinType;
+import org.apache.flink.table.runtime.operators.join.adaptive.AdaptiveJoin;
+
+import org.apache.flink.shaded.guava32.com.google.common.base.Preconditions;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/** The stream graph optimization strategy of adaptive broadcast join. */
+public class AdaptiveBroadcastJoinOptimizationStrategy
+        extends BaseAdaptiveJoinOperatorOptimizationStrategy {
+    private static final Logger LOG =
+            LoggerFactory.getLogger(AdaptiveBroadcastJoinOptimizationStrategy.class);
+
+    private Long broadcastThreshold;
+
+    private Map<Integer, Map<Integer, Long>> aggregatedProducedBytesByTypeNumberAndNodeId;
+
+    @Override
+    public boolean onOperatorsFinished(
+            OperatorsFinished operatorsFinished, StreamGraphContext context) {
+        initialize(context.getStreamGraph().getConfiguration());
+        visitDownstreamAdaptiveJoinNode(operatorsFinished, context);
+
+        return true;
+    }
+
+    @Override
+    protected void tryOptimizeAdaptiveJoin(
+            OperatorsFinished operatorsFinished,
+            StreamGraphContext context,
+            ImmutableStreamNode adaptiveJoinNode,
+            List<ImmutableStreamEdge> upstreamStreamEdges,
+            AdaptiveJoin adaptiveJoin) {
+        for (ImmutableStreamEdge upstreamEdge : upstreamStreamEdges) {
+            IntermediateDataSetID relatedDataSetID =
+                    context.getConsumedIntermediateDataSetId(upstreamEdge.getEdgeId());
+            long producedBytes =
+                    operatorsFinished.getResultInfoMap().get(upstreamEdge.getSourceId()).stream()
+                            .filter(
+                                    blockingResultInfo ->
+                                            blockingResultInfo.getResultId() == relatedDataSetID)
+                            .mapToLong(BlockingResultInfo::getNumBytesProduced)
+                            .sum();
+            aggregatedProducedBytesByTypeNumber(
```

Comment:
aggregatedProducedBytesByTypeNumber -> aggregatedInputBytesByTypeNumber

For the joinNode, it is input bytes instead of produced bytes.

This also applies to `aggregatedProducedBytesByTypeNumberAndNodeId`.